ArcGISRuntimeSDKQt_CppSamples/Maps/GenerateOfflineMapLocalBasemap/GenerateOfflineMapLocalBasemap.cpp

```diff
@@ -124,9 +124,9 @@ void GenerateOfflineMapLocalBasemap::generateMapByExtent(double xCorner1, double
   const QString tempPath = m_tempDir.path() + "/OfflineMap.mmpk";
   const QString dataPath = defaultDataPath() + "/ArcGIS/Runtime/Data/tpkx";
 
-  // connect to the signal for when the default parameters are generated
-  connect(m_offlineMapTask, &OfflineMapTask::createDefaultGenerateOfflineMapParametersCompleted,
-          this, [this, dataPath, tempPath](const QUuid&, GenerateOfflineMapParameters params)
+  // generate parameters
+  m_offlineMapTask->createDefaultGenerateOfflineMapParametersAsync(mapExtent).then(this,
+  [this, tempPath, dataPath](GenerateOfflineMapParameters params)
```

Comment:
Cant use const ref here because of the setter calls below

Reply:
It reads a bit odd, but it makes sense how it's used that way.